app/src/main/java/com/space/space__chat/presentation/ui/ChatUsersFragment.kt

```diff
@@ -27,28 +28,29 @@ class ChatUsersFragment : BaseFragment<FragmentChatUsersBinding, ChatUsersViewMo
             initRecyclerView(this)
             binding.chatSendBTN.setOnClickListener {
                 sendMessage(viewModel)
-
             }
         }
     }
 
     private fun initRecyclerView(viewModel: ChatUsersViewModel) {
         binding.chatFragmentRV.adapter = adapter
         showMessages(viewModel)
-
+    }
+    private fun filterMessages(messages: List<MessageModel>) : List<MessageModel> {
+        return messages.filter {
+            it.sender == adapterListener.invoke() || it.isOnline
+        }
     }
 
     private fun sendMessage(viewModel: ChatUsersViewModel) {
-        with(binding.chatInputET){
-            viewModel.sendMessage(text.toString(), tag.toString())
-            text?.clear()
+        with(viewModel) {
+            sendMessage(binding.chatInputET.text.toString(), tag.toString(),requireContext().isNetworkAvailable())
+            binding.chatInputET.text?.clear()
```

Comment:
u can write something like with(vm) { with(binding) {} }, in this case u you dont need it but i think you will need it

Reply:
აუ მეწერა ეგრე და nullს აგდებდა და ისევ დავაბრუნე ესე :)

Reply:
რაზე აგდებდა ნალს?

Reply:
მესიჯის გაგზავნისას ქრაშავდა და ნალს აგდებდა, ვერ მივხვდი რაუნდოდა და რატო და ესე დავაბრუნე და იყოს რა :დ